icu4c/source/common/restrace.h

```diff
@@ -66,6 +66,17 @@ class U_COMMON_API ResourceTracer {
     ~ResourceTracer();
 
     void trace(const char* type) const;
+    void traceOpen() const;
+
+    /**
+     * Calls trace() if the resB or parent provided to the constructor was
+     * non-null; otherwise, does nothing.
+     */
+    void traceOptional(const char* type) const {
```

Comment:
Optional: The suffix seems slightly odd. Ideas:
- maybeTrace(type)
- optionallyTrace(type)